src/Native/Runtime/MiscHelpers.cpp

```diff
@@ -535,6 +535,18 @@ COOP_PINVOKE_HELPER(void, RhSetThreadExitCallback, (void * pCallback))
 
 #endif // TARGET_UNIX
 
+#ifdef TARGET_WINDOWS
+
+// Function to call when a thread is detached from the runtime
+ThreadExitCallback g_threadExitCallback;
+
+COOP_PINVOKE_HELPER(void, RhSetThreadExitCallback, (void * pCallback))
+{
+    g_threadExitCallback = (ThreadExitCallback)pCallback;
```

Comment:
I'm sure this is not that straightforward and most likely I gonna need `FlsSetValue`. Just wanted to kick build. Going to read about fibers api.

Reply:
Somehow this appeared to work. To my understanding this callback being invoked in a static context (without any knowledge of parent fiber) and this is sufficient in our case. I haven't checked fibers docs yet, just was curious whether it gonna work with the most dumb approach.

Reply:
Seems nothing wrong with global callback registration/invokation although there is not much information in the internets.